File: Domain/Property.cs
using System;

namespace Domain
{
    public class Property
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Location { get; set; }
        public string Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }

        // Navigation Properties
        public virtual Photo Image { get; set; }

        //public virtual ICollection<Transaction> Transactions { get; set; } // Old
        // public string PropertyTypeId { get; set; } // Old
    }
}

File: client-app/src/app/models/Property.ts
export interface IProperty {
    id: string | null;
    name: string;
    description: string;
    location: string;
    image?: Image;
    status: string;
    propertyTypeId: string | null;
}

export interface IPropertyFormValues extends Partial<IProperty> {
}

export class PropertyFormValues implements IPropertyFormValues {
    id?: string = undefined;
    name: string = '';
    description: string = '';
    location: string = '';
    status: string = '';
    propertyTypeId?: string = undefined;
   
}

export interface Image {
    id: string;
    url: string;
    isMain: boolean;
} 